CONTRIBUTING.rst

```diff
@@ -0,0 +1,678 @@
+    configure the Travis CI GitHub App at
+    `Configure Travis CI <https://github.com/settings/installations>`__.
+
+-   Set repository access to either "All repositories" for convenience, or "Only
+    select repositories" and choose ``USERNAME/airflow`` in the drop-down menu.
+
+-   Access Travis CI for your fork at `<https://travis-ci.com/USERNAME/airflow>`__.
+
+Using Travis CI GitHub Services (legacy version)
+------------------------------------------------
+
+..
+
+    The apache/airflow project is still using the legacy version.
+
+Travis CI GitHub Services version uses an Authorized OAuth App.
+
+1.  Once installed, configure the Travis CI Authorized OAuth App at
+    `Travis CI OAuth APP <https://github.com/settings/connections/applications/88c5b97de2dbfc50f3ac>`__.
+
+2.  If you are a GitHub admin, click the **Grant** button next to your
+    organization; otherwise, click the **Request** button. For the Travis CI
+    Authorized OAuth App, you may have to grant access to the forked
+    ``ORGANIZATION/airflow`` repo even though it is public.
+
+3.  Access Travis CI for your fork at
+    `<https://travis-ci.org/ORGANIZATION/airflow>`_.
+
+Creating New Projects in Travis CI
+----------------------------------
+
+If you need to create a new project in Travis CI, use travis-ci.com for both
+private repos and open source.
+
+The travis-ci.org site for open source projects is now legacy and you should not use it.
+
+..
+    There is a second Authorized OAuth App available called "Travis CI for Open Source" used
+    for the legacy travis-ci.org service Don't use it for new projects.
+
+More information:
+
+-  `Open Source on travis-ci.com <https://docs.travis-ci.com/user/open-source-on-travis-ci-com/>`__.
+-  `Legacy GitHub Services to GitHub Apps Migration Guide <https://docs.travis-ci.com/user/legacy-services-to-github-apps-migration-guide/>`__.
+-  `Migrating Multiple Repositories to GitHub Apps Guide <https://docs.travis-ci.com/user/travis-migrate-to-apps-gem-guide/>`__.
+
+Changing the Metadata Database
+==============================
+
+When developing features, you may need to persist information to the metadata
+database. Airflow has `Alembic <https://bitbucket.org/zzzeek/alembic>`__. built-in
+module to handle all schema changes. Alembic must be installed on your
+development machine before continuing with migration.
+
+
+.. code-block:: bash
+
+    # starting at the root of the project
+    $ pwd
+    ~/airflow
+    # change to the airflow directory
+    $ cd airflow
+    $ alembic revision -m "add new field to db"
+       Generating
+    ~/airflow/airflow/migrations/versions/12341123_add_new_field_to_db.py
+
+
+Setting up the node / npm Javascript Environment
+================================================
+
+`airflow/www/` contains all npm-managed, front-end assets. Flask-Appbuilder
+itself comes bundled with jQuery and bootstrap. While they may be phased out
+over time, these packages are currently not managed with npm.
```

Comment:
Elena: //why is that setup needed? for what type of env? breeze or local virtualenv? shall we move it to the dev env section?

Jarek: This is for local environment. In breeze environment installing npm packages happens automatically. So yes it should be moved to LOCAL_VIRTUALENV.rst.